code/modules/core_implant/cruciform/machinery/eotp.dm

```diff
@@ -0,0 +1,194 @@
+var/global/obj/machinery/power/eotp/eotp
+
+#define ARMAMENTS "Armaments"
+#define ALERT "Antag Alert"
+#define INSPIRATION "Inspiration"
+#define ODDITY "Oddity"
+#define STAT_BUFF "Stat Buff"
+#define MATERIAL_REWARD "Materials"
+
+
+/obj/machinery/power/eotp
+	name = "Eye of the Protector"
+	desc = "He observe, he protects."
+	icon = 'icons/obj/eotp.dmi'
+	icon_state = "Eye_of_the_Protector"
+
+	density = TRUE
+	anchored = TRUE
+	layer = 5
+
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 30
+	active_power_usage = 2500
+
+	var/list/rewards = list(ARMAMENTS, ALERT, INSPIRATION, ODDITY, STAT_BUFF, MATERIAL_REWARD)
+
+	var/list/materials = list(/obj/item/stack/material/gold = 60,
+							/obj/item/stack/material/uranium = 30,
+							/obj/item/stack/material/plasma = 30,
+							/obj/item/stack/material/diamond = 30,
+							/obj/item/stack/material/plasteel = 120,
+							/obj/item/stack/material/silver = 60)
+	var/list/disk_types = list(/obj/item/weapon/computer_hardware/hard_drive/portable/design/nt_melee/longsword, /obj/item/weapon/computer_hardware/hard_drive/portable/design/nt_melee/scourge,
+							/obj/item/weapon/computer_hardware/hard_drive/portable/design/nt_grenades, /obj/item/weapon/computer_hardware/hard_drive/portable/design/nt_melee/halberd,
+							/obj/item/weapon/computer_hardware/hard_drive/portable/design/nt_melee/shield, /obj/item/weapon/computer_hardware/hard_drive/portable/design/armor/crusader,
+							/obj/item/weapon/computer_hardware/hard_drive/portable/design/nt_firstaid)
+
+	var/list/mob/living/carbon/human/scanned = list()
+	var/max_power = 100
+	var/power = 0
+	var/power_gaine = 2
+	var/max_observation = 800
+	var/observation = 0
+	var/min_observation = -100
+
+	var/stat_buff_power = 10
+
+	var/power_cooldown = 1 MINUTES
+	var/last_power_update = 0
+	var/rescan_cooldown = 10 MINUTES
+	var/last_rescan = 0
+
+/obj/machinery/power/eotp/New()
+	..()
+	eotp = src
+	START_PROCESSING(SSobj, src)
```

Comment:
1. this is a machine so it uses SSmachines, 2. I think this is unnecessary since all machines start SSmachines when initialized.

Reply:
Sure.

Reply:
Done.